Mage/src/main/java/mage/abilities/common/DealsDamageTriggeredAbility.java

```diff
@@ -0,0 +1,42 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+
+/**
+ * @author LevelX2
+ */
+
+public class DealsDamageTriggeredAbility extends TriggeredAbilityImpl {
```

Comment:
It was copy pasted from `DealsDamageGainLifeSourceTriggeredAbility`, but effect param. If you extracted that code then original ability must extends new one class. So code will be shared between both abilities.